aws/resource_aws_acm_certificate.go

```diff
@@ -0,0 +1,237 @@
+				ForceNew: true,
+			},
+			"subject_alternative_names": &schema.Schema{
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"validation_method": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
+					if v.(string) != "DNS" {
+						errors = append(errors, fmt.Errorf("only validation_method DNS is supported at the moment"))
+					}
+					return
+				},
+			},
+			"certificate_arn": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+				ForceNew: true,
+			},
+			"domain_validation_options": &schema.Schema{
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"domain_name": &schema.Schema{
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"resource_record_name": &schema.Schema{
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"resource_record_type": &schema.Schema{
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"resource_record_value": &schema.Schema{
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceAwsAcmCertificateCreate(d *schema.ResourceData, meta interface{}) error {
+	acmconn := meta.(*AWSClient).acmconn
+	params := &acm.RequestCertificateInput{
+		DomainName:       aws.String(d.Get("domain_name").(string)),
+		ValidationMethod: aws.String("DNS"),
+	}
+
+	sans, ok := d.GetOk("subject_alternative_names")
+	if ok {
+		sanStrings := sans.([]interface{})
+		params.SubjectAlternativeNames = expandStringList(sanStrings)
+	}
+
+	log.Printf("[DEBUG] ACM Certificate Request: %#v", params)
+	resp, err := acmconn.RequestCertificate(params)
+
+	if err != nil {
+		return fmt.Errorf("Error requesting certificate: %s", err)
+	}
+
+	d.SetId(*resp.CertificateArn)
+	if v, ok := d.GetOk("tags"); ok {
+		params := &acm.AddTagsToCertificateInput{
+			CertificateArn: resp.CertificateArn,
+			Tags:           tagsFromMapACM(v.(map[string]interface{})),
+		}
+		_, err := acmconn.AddTagsToCertificate(params)
+
+		if err != nil {
+			return fmt.Errorf("Error requesting certificate: %s", err)
+		}
+	}
+
+	return resourceAwsAcmCertificateRead(d, meta)
+}
+
+func resourceAwsAcmCertificateRead(d *schema.ResourceData, meta interface{}) error {
+	acmconn := meta.(*AWSClient).acmconn
+
+	params := &acm.DescribeCertificateInput{
+		CertificateArn: aws.String(d.Id()),
+	}
+
+	return resource.Retry(time.Duration(1)*time.Minute, func() *resource.RetryError {
+		resp, err := acmconn.DescribeCertificate(params)
+
+		if err != nil {
+			return resource.NonRetryableError(fmt.Errorf("Error describing certificate: %s", err))
+		}
+
+		if *resp.Certificate.Type != "AMAZON_ISSUED" {
```

Comment:
Mainly because I haven't tested it yet. I'm guessing `DomainValidationOptions` would be missing so similar to support for E-Mail validation we'd need to be more flexible with what data we expect from the API. 

For now, I fixed the the nitpick. For the rest, just like E-Mail validation, we have to make the decision to split work on this into a separate PR or work on it in this one.

Reply:
Done, removed the limitation on `AMAZON_ISSUED` certificates, `IMPORTED` certificates. 

Also added a warning to the docs that this should be used carefully as it might lead to fragile terraform projects: `IMPORTED` certificates can be imported and managed by terraform but not re-created after being destroyed.